scripts/session-wrapper.sh

```diff
@@ -3,6 +3,11 @@
 # This script runs outside of snap confinement as a wrapper around the
 # confined desktop session.
 
+systemctl --user enable pipewire.socket
+systemctl --user enable pipewire-pulse.socket
+systemctl --user start pipewire.socket
+systemctl --user start pipewire-pulse.socket
```

Comment:
Yes, I know. But I've been trying to make it work, and it's not so easy. Gnome Shell doesn't detect the audio, or pipewire-pulse fails to launch, or... a lot of things. That's why I think that maybe we should begin with the "easy" part, and step by step make it complete.

Reply:
Also, deleting those units from core22 isn't enough. I don't know why, but pipewire keeps being launched as a systemd user daemon over and over again, no matter what I delete from /etc/systemd, or ~/.config/systemd... so at least a "systemd --user stop pipewire.socket" and a "systemd --user stop pipewire.service" must be added there to avoid it being launched and interfere with the containerized one.

Reply:
And also, if pipewire is being launched inside the desktop session, then socket activation is not needed because gnome-shell itself will launch it as soon as it runs (because it need it for the "volume" control in the top bar).

Reply:
While delayed start is one benefit of socket activation, the main benefits are to correctly sequence start-up (i.e. you can start a service in parallel to of its dependencies, and it will block when it tries to connect to them), and to give continuity over crashes (i.e. the new instance of the service uses the same listening socket, and can accept connection attempts from after the first instance crashed).

As far as disabling the pipewire units in the base snap, we'd need to remove any pipewire units under `/usr/lib/systemd/user` as well as `/etc/systemd/user`. That should be enough to ensure only the units provided by the snap are used.